api/v3/Membership.php

```diff
@@ -33,6 +33,23 @@
  */
 
 /**
+ * Adjust Metadata for Delete action.
+ *
+ * The metadata is used for setting defaults, documentation & validation.
+ *
+ * @param array $params
+ *   Array of parameters determined by getfields.
+ */
+function _civicrm_api3_membership_delete_spec(&$params) {
+  $params['preserve_contribution'] = array(
+    'api.required' => 0,
+    'title' => 'Preserve Contribution',
+    'description' => 'By default this is 0, or 0 if not set. Set to 1 to preserve the associated contribution record when membership is deleted.',
```

Comment:
I guess I was saying I'd be happy to go further - ie. I'm saying the default should be preserve not delete - but at that point people who want to mess up peoples' databases have to change their code anyway - at which point I don't know if it's worth supporting delete-contribution-via-membership-api at all - they can do that in their code in the very unusual case where they really want to